language-api/src/test/java/de/jplag/ParsingExceptionTest.java

```diff
@@ -0,0 +1,23 @@
+package de.jplag;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+import java.io.File;
+
+import org.junit.jupiter.api.Test;
+
+public class ParsingExceptionTest {
+
+    @Test
+    void testParsingExceptionAcceptsNullReason() {
```

Comment:
Just a small note but the test method does not match its body because you are testing far more than only a null reason.
`testParsingExceptionConstructor` maybe?

Reply:
I can probably find a better name, but I initially stumbled upon this because I called the `ParsingException(File, Throwable)` constructor. It delegates to the `ParsingException(File, String, Throwable)` constructor with `null` for the string, so the NPE was thrown even when not passing a reason at all. I decided to call all public constructors in the test case to cover internal delegations, but I could also only test the `ParsingException(File, String, Throwable)` constructor instead.

Nullability of the other arguments isn't tested here, so `testParsingExceptionConstructor` might be too generic. Maybe `testParsingExceptionAcceptsNoOrNullReason`?

Reply:
Ah thanks for the clarification. Actually, I would say in such a case it is sufficient to call the 3 args constructor if for the other constructors internal delegation is performed anyways. But it doesn't harm to have all 4 calls here neither, so then just leave it as is (and also the method name)